kernel/src/transaction/mod.rs

```diff
@@ -253,14 +269,23 @@ fn generate_adds<'a>(
     let log_schema = get_log_add_schema();
 
     add_files_metadata.map(move |add_files_batch| {
-        let adds_expr = Expression::struct_from([Expression::struct_from(
-            add_files_schema
-                .fields()
-                .map(|f| Expression::column([f.name()])),
-        )]);
+        // Create a struct expression that transforms all fields
+        let field_expressions: Vec<Expression> = add_files_schema
+            .fields()
+            .map(|f| match f.name().as_str() {
+                // numRecords must be nested in stats and converted to JSON
+                "numRecords" => {
+                    Expression::unary(UnaryExpressionOp::ToJson, Expression::column([f.name()]))
+                }
```

Comment:
This expression does not work as expected yet, and after trying for a while, I couldn't figure out how to use the existing expressions to achieve the desired outcome.

The problem I see is that we need to transform `numRecords` into a nested struct before converting it to JSON. However, the schema of this intermediate struct does not appear in the output schema, so Kernel complains about not knowing the data types of the intermediate schema. The changed tests in this PR describe the desired output state.

Is there a way to solve this in a single expression, or do we have to apply multiple expressions in succession? I could not find a similar example in the code base, so I would appreciate help on how to transform `numRecords` into a nested `stats` struct, which is then converted into JSON.